src/sparseml/core/session.py

```diff
@@ -46,26 +56,53 @@ class _CallbackContainer:
 
 
 class SparseSession:
+    """
+    A session for sparsification that holds the current state of the sparsification
+    """
+
     def __init__(self):
         self._lifecycle = SparsificationLifecycle()
 
     @property
     def lifecycle(self) -> SparsificationLifecycle:
+        """
+        :return: the lifecycle for the session
+        """
         return self._lifecycle
 
     @property
     def state(self) -> State:
+        """
+        :return: the current state of the session
+        """
         return self._lifecycle.state
 
     def pre_initialize_structure(
         self,
         model: Any,
-        recipe: Union[str, List[str], Recipe, List[Recipe]] = None,
-        recipe_stage: Union[str, List[str]] = None,
-        recipe_args: Union[Dict[str, Any], List[Dict[str, Any]]] = None,
-        framework: Framework = None,
+        recipe: Union[str, List[str], Recipe, List[Recipe], None] = None,
+        recipe_stage: Union[str, List[str], None] = None,
+        recipe_args: Union[Dict[str, Any], List[Dict[str, Any]], None] = None,
+        framework: Optional[Framework] = None,
         **kwargs,
     ) -> ModifiedState:
+        """
+        A method to pre-initialize the structure of the model for sparsification.
+        This will run the pre-initialize structure method for each modifier in the
+        session's lifecycle. This will also set the session's state to the
+        pre-initialized state. Takes care of cases when the model(s) structure
+        has been previosuly modified by a modifier.
```

Comment:
This is probably a question for Mark because I'm a bit unclear on it too, at a high level what is the difference between what happens in pre_init and init?